tests/unit/inference/test_inference.py

```diff
@@ -557,3 +557,53 @@ def test(self, model_family, model_name, task):
                        ds_output["results"][task]["ppl"])
         #assert ds_time <= bs_time
         assert ppl_diff < 0.01
+
+
+@pytest.mark.inference
+@pytest.mark.parametrize("model_w_task",
+                         [("bigscience/bloom-560m",
+                           "text-generation")],
+                         ids=["bloom-560m"])
+class TestVariableBatchSizeCudaGraph(DistributedTest):
+    world_size = 1
+
+    def test(
+        self,
+        model_w_task,
+        dtype,
+        query,
+        inf_kwargs,
+        invalid_model_task_config,
+    ):
+        if invalid_model_task_config:
+            pytest.skip(invalid_model_task_config)
+
+        model, task = model_w_task
+        local_rank = int(os.getenv("LOCAL_RANK", "0"))
+
+        pipe = pipeline(task,
+                        model=model,
+                        device=torch.device(f"cuda:{local_rank}"),
+                        framework="pt")
+        if dtype == torch.half:
+            pipe.model.half()
+
+        pipe.model = deepspeed.init_inference(pipe.model,
+                                              mp_size=1,
+                                              dtype=dtype,
+                                              replace_with_kernel_inject=True,
+                                              replace_method='auto',
+                                              enable_cuda_graph=True)
+
+        # Run pipe test with increasing and then decreasing batch size
+        success = True
+        try:
+            pipe(query, batch_size=2)
+            pipe([query] * 4, batch_size=4)
+            pipe(query, batch_size=1)
```

Comment:
Let's use a 1-4-1 batch pattern here to make sure the b=1 graph is still valid after we've run/created the b=4 graph.